pipx/animate.py

```diff
@@ -59,12 +60,19 @@ def print_animation(
     period: float,
     animate_at_beginning_of_line: bool,
 ):
+    (term_cols, _) = shutil.get_terminal_size(fallback=(9999, 24))
     while not event.wait(0):
         for s in symbols:
             if animate_at_beginning_of_line:
-                cur_line = f"{s} {message}"
+                if len(message) <= term_cols - 3:
+                    cur_line = f"{s} {message}"
+                else:
+                    cur_line = f"{s} {message:.{term_cols-6}}..."
             else:
-                cur_line = f"{message}{s}"
+                if len(message) <= term_cols - 4:
+                    cur_line = f"{message}{s}"
+                else:
```

Comment:
Also I think for this to not suffer easily regressions going ahead we should add some tests that validate the change.

Reply:
Agree, but I'm frankly not sure how to do that with animated (transient) text.  Anybody have a pointer or ideas?

Reply:
when the pytest capsys is attached the override does not happen, allowing you to check byte-wise; see https://github.com/tox-dev/tox/blob/master/tests/unit/util/test_spinner.py#L60-L81 for example

Reply:
Nice.  And the test above it is testing with animation.

Reply:
Another tidbit I just figured out, is that by setting env variables `COLUMNS` and `ROWS` you can make `shutil.get_terminal_size()` return values at your command.